sympy/solvers/solveset.py

```diff
@@ -1142,13 +1142,146 @@ def _is_exponential(f, symbol):
     return False
 
 
-def _transolve(f, symbol, domain):
+def _solve_log(f, symbol):
+    r"""
+    Helper to solve logarithmic equations.
+
+    Parameters
+    ==========
+
+    f : Expr
+        The logarithmic equation to be solved
+
+    symbol : Symbol
+        The variable in which the equation is solved
+
+    Returns
+    =======
+
+    An improved equation containg a single instance of log.
+
+    `None`:
+        If the equation does not changes.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, S, log
+    >>> from sympy.solvers.solveset import _solve_log as solve_log
+    >>> x = symbols('x')
+    >>> f = log(x - 3) + log(x + 3)
+    >>> solve_log(f, x)
+    log((x - 3)*(x + 3))
+
+    * Proof of correctness
+
+    A logarithm is another way to write exponent and is defined by
+
+    .. math:: {log_b x} = y \enspace if \enspace b^y = x
+
+    When one side of the equation contains a single logarithm, the
+    equation can be solved by rewriting the equation as an equivalent
+    exponential equation as defined above. But if one side contains
+    more than one logarithm, we need to use the properties of logarithm
+    to condense it into a single logarithm.
+
+    Take for example
+
+    .. math:: log(2*x) - 15 = 0
+
+    contains single logarithm, therefore we can directly rewrite it to
+    exponential form as
+
+    .. math:: x = exp(15)/2
+
+    But if the equation has more than one logarithm as
+
+    .. math:: log(x - 3) + log(x + 3) = 0
+
+    we use log identities to convert into a reduced form
+
+    Using,
+
+    .. math:: log(a) + log(b) = log(a*b)
+
+    therefore the equation becomes,
+
+    .. math:: log((x - 3)*(x  + 3))
+
+    This equation contains one logarithm and can be solved by rewriting
+    to exponents.
+    """
+
+    lhs, rhs_s = invert_real(f, 0, symbol)
+    rhs = rhs_s.args[0]
+    new_f = None
+
+    new_lhs = logcombine(lhs, force=True)
```

Comment:
Yeah you are right but it seems to me using `force=True` is a necessity, otherwise equations won't get simplified like take for eg: `log(x-3) + log(x+3)`, even if x is assumed to be positive using just `logcombine` would not simplify it to `log((x-3)*(x+3))` because arguments `x - 3` and `x + 3` canot be assumed to be positive.
One thing that I think can be done is to use `force=True` and later we should check if this has imparted any kind of bad solutions (may be we can use `checksol`). Another thing which can be done is to modify the assumptions of the `free_symbols` internally (I guess it would be the optimal way)
I used `checksol` before but found a problem in it (as mentioned in the google groups discussion), I am not sure as of now that is why I have not included any type of check that would remove bad solutions. What are your thoughts on this?

Reply:
>It seems that if one of them is positive and the other negative that the same result will be obtained: log(neg)+log(pos) = log(neg*pos)

This result is valid if arguments are numbers, this is because `log(neg)` = `log(|neg|) + I*pi`

```python
>>> f = log(2) + log(-3)       # log(2) + log(3) + I*pi
>>>  logcombine(f)
log(6) + I*pi

>>> x = symbols('x', positive=True); y = symbols('y', negative=True)
>>> f = log(x) + log(y)
>>> logcombine(f)
log(x) + log(y)
```

> If checksol is used to ...

Okay so we should not use `checksol`. So how we will remove the singularities? Should we check each argument of every log term being real?